src/Arcus.Security.Tests.Unit/KeyVault/KeyVaultSecretProviderTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Arcus.Security.Core.Exceptions;
+using Arcus.Security.KeyVault;
+using Arcus.Security.KeyVault.Factories;
+using Microsoft.Azure.KeyVault;
+using Xunit;
+
+namespace Arcus.Security.Tests.Unit.KeyVault
+{
+    public class KeyVaultSecretProviderTests
+    {
+        [Fact]
```

Comment:
We're testing different things here so should be separate tests imo

Reply:
What do you mean?  I have too much assertions in one test?

Reply:
Not too many asserts but testing different scenarios. Anyway, jumped in and helped